plugins/MauticCitrixBundle/Form/Type/CitrixCampaignActionType.php

```diff
@@ -72,21 +73,23 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
         $builder->add(
             'event-criteria-'.$product,
-            'choice',
+            ChoiceType::class,
             [
-                'label'   => $this->translator->trans('plugin.citrix.action.criteria'),
-                'choices' => $newChoices,
+                'label'             => $this->translator->trans('plugin.citrix.action.criteria'),
+                'choices'           => array_flip($newChoices),
```

Comment:
You van flip label and id in foreach on line 78

Reply:
And use in_array on line 97, 98, 99

Reply:
Isn't it easier to change only 1 line instead of 4 lines?
In your version we get rid of the `array_flip`, but we have to change 3 other lines of the code (replace array_key_exists with in_array).
What is the advantage of your proposition?

Reply:
Get rid of array_flip function call. Speed improvement.

Reply:
I was just curious maybe there are some other thoughts.
http://maettig.com/1397246220
It's unclear if there will be any speed improvement if we get rid of `array_flip` because `in_array` is slower than `array_key_exists`.
However I don't want too discuss this too much so I've refactored it as you said.
Fixed in 70a2d96be4bc420d930bebe4751ec90dd680bfe2